modules/programs/mu.nix

```diff
@@ -0,0 +1,86 @@
+
+with lib;
+
+let
+
+  cfg = config.programs.mu;
+
+  # By default, mu will place it's database in XDG_CACHE_HOME/mu.
+  defaultDBLocation = config.xdg.cacheHome + "/mu";
+
+  # Used to generate command line arguments that mu can operate with.
+  genCmdMaildir = path: "--maildir=" + path;
+
+  genCmdAddrs = addrs:
+    let
+      # Takes the list of address strings and returns list of strings with the
+      # --my-address= prefixed to each
+      addMyAddr = addrs: builtins.map (addr: "--my-address=" + addr) addrs;
+    in builtins.concatStringsSep " " (addMyAddr addrs);
+
+in {
+  meta.maintainers = [ maintainers.KarlJoad ];
+
+  options.programs.mu = {
+    enable = mkEnableOption "mu, a maildir indexer and searcher";
+    # No options/config file present for mu, and program author will not be adding
+    # one soon.
+    # See https://github.com/djcb/mu/issues/882 for more information about this.
+
+    # All configuration options below affect only the initial index script wrapper.
+
+    maildir = mkOption {
+      type = types.path;
+      default = config.accounts.email.maildirBasePath;
+      defaultText = "$HOME/Maildir";
+      example = "$HOME/Mail";
+      description = ''
+        Path to the base directory for all accounts' maildir directories. May be
+        a relative path or absolute path. Must be specified for the initial index,
+        then, never again.
+
+        Note that if you define your own path, you cannot leave a trailing
+        <literal>/</literal>.
+      '';
+    };
+
+    addresses = mkOption {
+      type = types.listOf types.str;
+      default = builtins.attrValues
+        (builtins.mapAttrs (account: vals: vals.address)
+          config.accounts.email.accounts);
+      defaultText = [
+        "accounts.email.accounts.<name1>.address"
+        "accounts.email.accounts.<name2>.address"
+        "..."
+      ];
+      example = [ "user1@domain1.com" "user2@domain2.net" ];
+      description = ''
+        List of email accounts that mu should initialize the maildir store and
+        xapian database with, and subsequently index with.
+        By default, the list will contain all email accounts specified in
+        <literal>accounts.email.accounts</literal>.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ pkgs.mu ];
+
+    home.activation.runMuInit = let
+      maildirOption = genCmdMaildir cfg.maildir;
+      addrsOption = genCmdAddrs cfg.addresses;
+    in hm.dag.entryAfter [ "writeBoundary" ] ''
+      # If the database directory exists, then `mu init` should NOT be run.
+      # In theory, mu is the only thing that creates that directory, and it is
+      # only created during the initial index.
+      if [[ ! -d "${cfg.dbLocation}" ]]; then
+        $DRY_RUN_CMD mu init ${maildirOption} ${addrsOption} $VERBOSE_ARG;
+      fi
+
+      # mu uses MAILDIR to find user's Maildir if not specified explicitly at init
+      export MAILDIR="${cfg.maildir}"
```

Comment:
I don't understand the function of this. Are there additional calls to `mu` after this line that would pick up the variable? Isn't the point that the directory is set in the `mu init` call?

Reply:
This is for the case where you have placed the `maildir` at a non-default location. From the previous comment, we can change the `config.accounts.email.maildirBasePath` to something non-default (default as assumed from `mu`). However, `mu` will first check against the `MAILDIR` environment variable to find the mail directory. If `maildirBasePath` was changed, then this needs to be set.

Essentially, `mu init` with a maildir path is not persistent.

Reply:
I stand correct it. I was wrong. It's been a while since I worked on this module.

There is no justifiable reason for it to be there that I can think of.